test/parallel/test-inspector-async-hook-after-done.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+
+const common = require('../common');
+
+common.skipIfInspectorDisabled();
+
+const { Worker } = require('worker_threads');
+const { Session } = require('inspector');
+
+const session = new Session();
+
+session.connect();
+
+session.on('NodeWorker.attachedToWorker', async ({ params: { sessionId } }) => {
+  let id = 1;
+  function postToWorker(method, params) {
+    session.post('NodeWorker.sendMessageToWorker', {
+      sessionId,
+      message: JSON.stringify({ id: id++, method, params })
+    });
+  }
+
+  // Wait for the notification
+  function onMessageReceived({ params: { message } }) {
+    if (!message ||
+      JSON.parse(message).method !== 'NodeRuntime.waitingForDisconnect') {
+      session.once('NodeWorker.receivedMessageFromWorker', onMessageReceived);
+      return;
+    }
+    // Force a call to node::inspector::Agent::ToggleAsyncHook by changing the
+    // async call stack depth
+    postToWorker('Debugger.setAsyncCallStackDepth', { maxDepth: 1 });
+    // This is were the original crash happened
+    session.post('NodeWorker.detach', { sessionId });
```

Comment:
Should we await calling callback of this function?

Reply:
Thanks for pointing this out. Note that for this test to run to a completion we need to have async handle attached to a main loop - so I use timers.